src/components/dashboard/dashboardItem/LoginTable.js

```diff
@@ -0,0 +1,84 @@
+/**
+ *
+ * @author mian
+ *
+ * A table of recent logins for the user.
+ *
+ */
+
+import React from "react";
+import { useQuery } from "react-query";
+
+import { logins, LOGINS_QUERY_KEY } from "serviceFacades/users";
+import ErrorTypographyWithDialog from "components/error/ErrorTypographyWithDialog";
+import { useTranslation } from "i18n";
+import {
+    formatDate,
+    getFormattedDistance,
+} from "components/utils/DateFormatter";
+import {
+    useTheme,
+    Skeleton,
+    Table,
+    TableBody,
+    TableCell,
+    TableContainer,
+    TableHead,
+    TableRow,
+    Paper,
+} from "@mui/material";
+
+export default function LoginsTable() {
+    const { t } = useTranslation("dashboard");
+    const { t: i18nCommon } = useTranslation("common");
+
+    const theme = useTheme();
+    const { status, data, error } = useQuery({
+        queryKey: [LOGINS_QUERY_KEY],
+        queryFn: () => logins({ limit: 5 }),
+    });
+
+    if (status === "error") {
+        return (
+            <div style={{ padding: theme.spacing(1) }}>
+                <ErrorTypographyWithDialog
+                    errorObject={error}
+                    errorMessage={""}
+                />
+            </div>
+        );
+    }
+    if (status === "loading") {
+        return <Skeleton variant="rectangular" height={200} />;
+    }
+    return (
+        <TableContainer component={Paper}>
+            <Table size="small">
+                <TableHead>
+                    <TableRow>
+                        <TableCell>{t("loginTime")}</TableCell>
+                        <TableCell></TableCell>
+                        <TableCell>{t("ipAddress")}</TableCell>
+                    </TableRow>
+                </TableHead>
+                <TableBody>
+                    {data["logins"].map((row) => (
+                        <TableRow>
+                            <TableCell>
+                                {i18nCommon("timestamp", {
+                                    timestamp: getFormattedDistance(
+                                        row["login_time"] / 1000
+                                    ),
+                                })}
+                            </TableCell>
+                            <TableCell>
+                                ({formatDate(row["login_time"])})
+                            </TableCell>
+                            <TableCell>{row["ip_address"]}</TableCell>
```

Comment:
Just a reminder that accessing JSON objects like arrays work fine (and is required if the field has characters like `-`), but you can also access these fields as `row.ip_address`, `row.login_time`, and `data?.logins` (best to guard with `?` on `data` in case it didn't load yet, although the previous `error` and `loading` checks should guard against that as well).

Reply:
I've kept it as array-like syntax for now for the individual row fields, but I did change it over to the `data?.logins` way, since that did seem like a good point. I figure that the array-like syntax lets it be consistent when there's hyphens, but that's probably not really very meaningful, so if anyone feels strongly about it I can change it to dot notation.